app/lib/features/events/pages/event_list_page.dart

```diff
@@ -0,0 +1,224 @@
+import 'dart:math';
+
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/providers/space_providers.dart';
+import 'package:acter/common/toolkit/buttons/primary_action_button.dart';
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/common/widgets/add_button_with_can_permission.dart';
+import 'package:acter/common/widgets/empty_state_widget.dart';
+import 'package:acter/features/events/providers/event_providers.dart';
+import 'package:acter/features/events/widgets/event_item.dart';
+import 'package:acter/features/events/widgets/skeletons/event_list_skeleton_widget.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:atlas_icons/atlas_icons.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_staggered_grid_view/flutter_staggered_grid_view.dart';
+import 'package:go_router/go_router.dart';
+import 'package:acter/features/search/providers/search.dart';
+
+class EventListPage extends ConsumerStatefulWidget {
+  final String? spaceId;
+
+  const EventListPage({super.key, this.spaceId});
+
+  @override
+  ConsumerState<EventListPage> createState() => _EventListPageState();
+}
+
+class _EventListPageState extends ConsumerState<EventListPage> {
+  final TextEditingController searchTextController = TextEditingController();
+
+  String get searchValue => ref.watch(searchValueProvider);
+
+  EventFilters get eventFilterValue => ref.watch(eventFilerProvider);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: _buildAppBar(),
+      body: _buildBody(),
+    );
+  }
+
+  AppBar _buildAppBar() {
+    return AppBar(
+      centerTitle: false,
+      title: Column(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(L10n.of(context).events),
+          if (widget.spaceId != null) _buildSpaceName(),
+        ],
+      ),
+      actions: [
+        AddButtonWithCanPermission(
+          canString: 'CanPostEvent',
+          onPressed: () => context.pushNamed(
+            Routes.createEvent.name,
+            queryParameters: {'spaceId': widget.spaceId},
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildSpaceName() {
+    String spaceName =
+        ref.watch(roomDisplayNameProvider(widget.spaceId!)).valueOrNull ?? '';
+    return Text(
+      '($spaceName)',
+      overflow: TextOverflow.ellipsis,
+      style: Theme.of(context).textTheme.labelLarge,
+    );
+  }
```

Comment:
Don't you think it's time we put that into a separate commons widget now?

Reply:
Yes true, let me do it